### ISSUE TITLE
EventsManager.EnterpriseEventsAsync() generates an ill-formed created_after parameter.

### ISSUE THREAD

**Issue by thchuong-x11** [box/box-windows-sdk-v2#561]
- [X ] I have checked that the [SDK documentation][sdk-docs] and [API documentation][api-docs] doesn't solve my issue

### Description of the Issue
For monitoring purpose, I use the Enterprise Events endpoint to retrieve the events that occur on my box. After updating to the version v3.16.0, the API stops to work and I have this kind of error:
```
System.AggregateException: One or more errors occurred. ---> Box.V2.Exceptions.BoxException: The API returned an error [BadRequest | rsjus8g2ruhtq2nq.0u9akr8m4qfs9fnp5snhn0insjb] bad_request - Invalid created_after parameter
   at Box.V2.Extensions.BoxResponseExtensions.ParseResults[T](IBoxResponse`1 response, IBoxConverter converter) in Box.V2\Extensions\BoxResponseExtensions.cs:line 83
   at Box.V2.Managers.BoxResourceManager.<ToResponseAsync>d__11`1.MoveNext() in ...\Box.V2\Managers\BoxResourceManager.cs:line 77
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Box.V2.Managers.BoxEventsManager.<EnterpriseEventsAsync>d__3.MoveNext() in ...\Box.V2\Managers\BoxEventsManager.cs:line 64
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at Sinequa.Connectors.BoxCom.TestClass.TestEnterpriseEvents() in \TestClass.cs:line 62
---> (Inner Exception #0) Box.V2.Exceptions.BoxException: The API returned an error [BadRequest | rsjus8g2ruhtq2nq.0u9akr8m4qfs9fnp5snhn0insjb] bad_request - Invalid created_after parameter
   at Box.V2.Extensions.BoxResponseExtensions.ParseResults[T](IBoxResponse`1 response, IBoxConverter converter) in ...\Box.V2\Extensions\BoxResponseExtensions.cs:line 83
   at Box.V2.Managers.BoxResourceManager.<ToResponseAsync>d__11`1.MoveNext() in ...\Box.V2\Managers\BoxResourceManager.cs:line 77
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
```

here is the request that is generated by v3.12.0 (first) vs. v3.16.0 (second)
v3.12.0
`https://api.box.com/2.0/events?stream_type=admin_logs&limit=500&event_type=COPY,DELETE,EDIT,MOVE,RENAME,UNDELETE,UPLOAD,COLLABORATION_ACCEPT,COLLABORATION_ROLE_CHANGE,COLLABORATION_REMOVE,COLLABORATION_INVITE,UPDATE_COLLABORATION_EXPIRATION,COLLABORATION_EXPIRATION,TASK_ASSIGNMENT_UPDATE,TASK_ASSIGNMENT_CREATE,TASK_ASSIGNMENT_DELETE,TASK_CREATE,TASK_UPDATE,COMMENT_CREATE,COMMENT_DELETE,METADATA_INSTANCE_CREATE,METADATA_INSTANCE_UPDATE,METADATA_INSTANCE_DELETE&created_after=2019-05-06T11:09:33.8120000Z`
v3.16.0
`https://api.box.com/2.0/events?stream_type=admin_logs&limit=500&event_type=COPY%2CDELETE%2CEDIT%2CMOVE%2CRENAME%2CUNDELETE%2CUPLOAD%2CCOLLABORATION_ACCEPT%2CCOLLABORATION_ROLE_CHANGE%2CCOLLABORATION_REMOVE%2CCOLLABORATION_INVITE%2CUPDATE_COLLABORATION_EXPIRATION%2CCOLLABORATION_EXPIRATION%2CTASK_ASSIGNMENT_UPDATE%2CTASK_ASSIGNMENT_CREATE%2CTASK_ASSIGNMENT_DELETE%2CTASK_CREATE%2CTASK_UPDATE%2CCOMMENT_CREATE%2CCOMMENT_DELETE%2CMETADATA_INSTANCE_CREATE%2CMETADATA_INSTANCE_UPDATE%2CMETADATA_INSTANCE_DELETE&created_after=2019-05-06T11%253A09%253A33.8120000Z`

I have verified with postman that the second request causes the same error and that by "URL-decoding" the `created_after` parameter, it works again.

### Versions Used
.NET SDK: v3.16.0
.NET Framework: 4.6

### Steps to Reproduce
Here is my code snippet to reproduce the problem:
```
internal static void TestEnterpriseEvents()
{
    const string clientId = "tde7641gqbtnj48868nbltf8fxaven3u";
    const string clientSecret = "";
    const string enterpriseId = "519981";
    const string jwtPrivateKey =
        @"";
    const string jwtPassphrase = "";
    const string jwtPublicKeyId = "vf3hat4l";
    const bool suppressNotifications = false;
    string[] eventTypes =
    {
        "COPY","DELETE"
    };

    var boxConfig = new Box.V2.Config.BoxConfig(clientId, clientSecret, enterpriseId, jwtPrivateKey, jwtPassphrase, jwtPublicKeyId);
    var boxJWTAuth = new Box.V2.JWTAuth.BoxJWTAuth(boxConfig);
    var adminToken = boxJWTAuth.AdminToken();
    var boxClient = boxJWTAuth.AdminClient(adminToken, null, suppressNotifications);

    var ok = true;
    try
    {
        var eventsTask = boxClient.EventsManager.EnterpriseEventsAsync(500, null, eventTypes, DateTime.Now);
        eventsTask.Wait();
        Console.WriteLine($"Fetched {eventsTask.Result.Entries.Count} events.");
    }
    catch (Exception e)
    {
        Console.WriteLine(e);
        ok = false;
    }

    Console.WriteLine(ok ? "Succeeds." : "Failed.");
}
```

### Error Message, Including Stack Trace
```
System.AggregateException: One or more errors occurred. ---> Box.V2.Exceptions.BoxException: The API returned an error [BadRequest | rsjus8g2ruhtq2nq.0u9akr8m4qfs9fnp5snhn0insjb] bad_request - Invalid created_after parameter
   at Box.V2.Extensions.BoxResponseExtensions.ParseResults[T](IBoxResponse`1 response, IBoxConverter converter) in Box.V2\Extensions\BoxResponseExtensions.cs:line 83
   at Box.V2.Managers.BoxResourceManager.<ToResponseAsync>d__11`1.MoveNext() in ...\Box.V2\Managers\BoxResourceManager.cs:line 77
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
   at Box.V2.Managers.BoxEventsManager.<EnterpriseEventsAsync>d__3.MoveNext() in ...\Box.V2\Managers\BoxEventsManager.cs:line 64
   --- End of inner exception stack trace ---
   at System.Threading.Tasks.Task.ThrowIfExceptional(Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task.Wait(Int32 millisecondsTimeout, CancellationToken cancellationToken)
   at System.Threading.Tasks.Task.Wait()
   at Sinequa.Connectors.BoxCom.TestClass.TestEnterpriseEvents() in \TestClass.cs:line 62
---> (Inner Exception #0) Box.V2.Exceptions.BoxException: The API returned an error [BadRequest | rsjus8g2ruhtq2nq.0u9akr8m4qfs9fnp5snhn0insjb] bad_request - Invalid created_after parameter
   at Box.V2.Extensions.BoxResponseExtensions.ParseResults[T](IBoxResponse`1 response, IBoxConverter converter) in ...\Box.V2\Extensions\BoxResponseExtensions.cs:line 83
   at Box.V2.Managers.BoxResourceManager.<ToResponseAsync>d__11`1.MoveNext() in ...\Box.V2\Managers\BoxResourceManager.cs:line 77
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1.ConfiguredTaskAwaiter.GetResult()
```

**Comment by mattwiller** [comment]
@thchuong-x11 Thank you for letting us know! We recently made a fix to URL encode all query parameters in the SDK, but it looks like there were a couple places (including `.EnterpriseEventsAsync()`) where parameters were already being URL encoded as one-offs.  This resulted in those query parameter values being encoded twice, which is the cause of the issue.  This should be pretty easy to fix (by just removing the one-off encodings); I'll get it done as soon as possible.

**Comment by mattwiller** [comment]
@thchuong-x11 The fix should be released now in v3.17.0 of the SDK; please give it a try and let us know if you run into any other issues!
